src/app.rs

```diff
@@ -113,13 +114,15 @@ impl AppData {
     }
 }
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Derivative, Serialize, Deserialize)]
+#[derivative(Debug)]
 pub struct Channel {
     /// Either phone number or group id
     pub id: String,
     pub name: String,
     pub is_group: bool,
-    pub messages: Vec<Message>,
+    #[derivative(Debug = "ignore")]
+    pub messages: StatefulList<Message>,
```

Comment:
Ah, good approach. Messages can have a state now.

I am just thinking about being backwards compatible in the storage format. I guess this serializes messages differently now, right? In JSON instead of 

```
{
  "messages": [{...}]
}
```

we got

```
{
  "messages": {
    "items": [{...}]
  }
}
```

If this is the case, we would loose the messages. Could you please confirm if this is like that?

If yes, the solution might be quite simple by using custom ser/de functions that flatten `items`.

Reply:
Yes, I can confirm that there is an additional layer of "items" for messages in the json format, and that this results in the old gurk.data.json file getting completely overwritten when switching from the old format to the new. It is good you caught that bug, losing all the old messages is a bad one. My return to work on this feature followed not having used gurk regularly for a while so I must have missed that this had happened with my own instance.

I also followed your suggestion of implementing custom serde functions to maintain backwards compatibility of the json storage format. I have tested and it seems to be working well! The format of the scroll_messages json now matches the original format, and no messages are lost when loading an old format json file into the new code.